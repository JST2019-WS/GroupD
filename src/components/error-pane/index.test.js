/* eslint-env node, jest */

import { h } from "preact";
import ErrorPane from "./index";
import { shallow } from 'enzyme'

describe('error-pane', () => {
    it('should render null errors', () => {
        let component = shallow(<ErrorPane error={null}/>);
        expect(component.html()).toMatchSnapshot('null-error-with-callback');

        component = shallow(<ErrorPane error={null} refreshCallback={() => {}}/>);
        expect(component.html()).toMatchSnapshot('null-error-without-callback');
    });

    it('should render ordinary errors', () => {
        let component = shallow(<ErrorPane error={new Error('Test')}/>);
        expect(component.html()).toMatchSnapshot('orindary-error');
    });

    it('should not render refresh if no callback is passed', () => {
        const component = shallow(<ErrorPane error={'Test'}/>);
        expect(component.html()).not.toContain('Retry');
    });
});
