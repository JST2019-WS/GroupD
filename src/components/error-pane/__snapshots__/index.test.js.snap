// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`null-error-with-callback 1`] = `"<div class=\\"dialog-pane-background\\"><div class=\\"dialog-pane-dialog\\" style=\\"background-color: darkred;\\">Unknown error<br /></div></div>"`;

exports[`null-error-without-callback 1`] = `"<div class=\\"dialog-pane-background\\"><div class=\\"dialog-pane-dialog\\" style=\\"background-color: darkred;\\">Unknown error<br /><div class=\\"error-pane-dialog__button\\">Retry</div></div></div>"`;

exports[`orindary-error 1`] = `"<div class=\\"dialog-pane-background\\"><div class=\\"dialog-pane-dialog\\" style=\\"background-color: darkred;\\"><undefined></undefined><br /></div></div>"`;
